lib/rules/no-unsupported-features/es-syntax.js

```diff
@@ -4,428 +4,40 @@
-                supported: "10.0.0",
-                messageId: "no-json-superset",
-            },
-        ],
-    },
-    optionalCatchBinding: {
-        ruleId: "no-optional-catch-binding",
-        cases: [
-            {
-                supported: "10.0.0",
-                messageId: "no-optional-catch-binding",
-            },
-        ],
-    },
-
-    //--------------------------------------------------------------------------
-    // ES2020
-    //--------------------------------------------------------------------------
-    bigint: {
-        ruleId: "no-bigint",
-        cases: [
-            {
-                supported: "10.4.0",
-                test: info => info.node.type === "Literal",
-                messageId: "no-bigint",
-            },
-            {
-                supported: null,
-                test: ({ node }) =>
-                    node.type === "Literal" &&
-                    (node.parent.type === "Property" ||
-                        node.parent.type === "MethodDefinition") &&
-                    !node.parent.computed &&
-                    node.parent.key === node,
-                messageId: "no-bigint-property-names",
-            },
-        ],
-    },
-    dynamicImport: {
-        ruleId: "no-dynamic-import",
-        cases: [
-            {
-                supported: new Range("^12.17 || >=13.2"),
-                messageId: "no-dynamic-import",
-            },
-        ],
-    },
-    optionalChaining: {
-        ruleId: "no-optional-chaining",
-        cases: [
-            {
-                supported: "14.0.0",
-                messageId: "no-optional-chaining",
-            },
-        ],
-    },
-    nullishCoalescingOperators: {
-        ruleId: "no-nullish-coalescing-operators",
-        cases: [
-            {
-                supported: "14.0.0",
-                messageId: "no-nullish-coalescing-operators",
-            },
-        ],
-    },
-
-    //--------------------------------------------------------------------------
-    // ES2021
-    //--------------------------------------------------------------------------
-    logicalAssignmentOperators: {
-        ruleId: "no-logical-assignment-operators",
-        cases: [
-            {
-                supported: "15.0.0",
-                messageId: "no-logical-assignment-operators",
-            },
-        ],
-    },
-    numericSeparators: {
-        ruleId: "no-numeric-separators",
-        cases: [
-            {
-                supported: "12.5.0",
-                messageId: "no-numeric-separators",
-            },
-        ],
-    },
-}
-const keywords = Object.keys(features)
+/**
+ * @type {{ruleId: string, supported: import("semver").Range, deprecated: boolean}[]}
+ */
+const features = JSON.parse(readFileSync(resolve(__dirname, "es-syntax.json")))
+/** @type {Set<string>} */
+const ignoreKeys = new Set()
+const ruleMap = Object.entries(features).map(([ruleId, meta]) => {
+    const ruleIdNegated = ruleId.slice(3)
```

Comment:
`.slice(3)` was to remove the `"no-"`?  good to have a comment for explanation here. :)